server/base/src/main/java/org/apache/accumulo/server/master/state/MetaDataStateStore.java

```diff
@@ -58,57 +54,31 @@ protected MetaDataStateStore(ClientContext context, CurrentState state, String t
     return new MetaDataTableScanner(context, TabletsSection.getRange(), state, targetTableName);
   }
 
-  @Override
   public void setLocations(Collection<Assignment> assignments) throws DistributedStoreException {
-    BatchWriter writer = createBatchWriter();
-    try {
+    try (var tabletsMutator = ample.mutateTablets()) {
       for (Assignment assignment : assignments) {
-        Mutation m = new Mutation(assignment.tablet.toMetaRow());
-        assignment.server.putLocation(m);
-        assignment.server.clearFutureLocation(m);
-        SuspendingTServer.clearSuspension(m);
-        writer.addMutation(m);
+        TabletMutator tabletMutator = tabletsMutator.mutateTablet(assignment.tablet);
+        tabletMutator.putLocation(assignment.server, LocationType.CURRENT);
+        tabletMutator.deleteLocation(assignment.server, LocationType.FUTURE);
+        tabletMutator.mutate();
```

Comment:
Did you forget to include the `deleteSuspension()` here?

Also, the code might look cleaner if you chained the methods, something like (not formatted):

```suggestion
        tabletsMutator.mutateTablet(assignment.tablet)
          .putLocation(assignment.server, LocationType.CURRENT)
          .deleteLocation(assignment.server, LocationType.FUTURE)
          .deleteSuspension()
          .mutate();
```

If the formatter makes it look too ugly, you could prevent the formatter from messing with it by wrapping it with a comment line before and after, `// @formatter:off` and `// @formatter:on`.

This kind of simplification can be done in other places in this PR also (I made a review comment elsewhere where it can be a one-liner.

Reply:
Yes, that was an oversight on my part. Added it too it and I will test how that looks.

Reply:
I'm going to go ahead and chain the methods in the other state stores and various files that use tabletMutator.